src/inference/src/ie_core.cpp

```diff
@@ -587,12 +587,15 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
 
         FOREACH_CHILD (pluginNode, devicesNode, "plugin") {
             std::string deviceName = GetStrAttr(pluginNode, "name");
-            ov::util::FilePath pluginPath = getPluginPath(GetStrAttr(pluginNode, "location"));
-
+            if (pluginRegistry.find(deviceName) != pluginRegistry.end()) {
+                IE_THROW() << "Device with \"" << deviceName << "\"  is already registered in the OpenVINO Runtime";
```

Comment:
can we add a test for this scenario?
```
Core core;
core.register_plugins("plugins.xml"); // <== must throw an exception
```

Reply:
Yes, done for both `register_plugin()` and `register_plugins()`